src/Data/List/Properties.agda

```diff
@@ -106,6 +107,15 @@ map-compose : {g : B → C} {f : A → B} → map (g ∘ f) ≗ map g ∘ map f
 map-compose []       = refl
 map-compose (x ∷ xs) = cong (_ ∷_) (map-compose xs)
 
+map-injective : ∀ (f : A ↣ B) → Injective _≡_ _≡_ (map (Injection.f f))
+map-injective f {[]}     {[]}     refl = refl
+map-injective f {x ∷ xs} {y ∷ ys} eq with ∷-injective eq
+... | fx≡fy , fxs≡fys = cong₂ _∷_ (injective fx≡fy) (map-injective f fxs≡fys)
+  where open Injection f hiding (f)
+
+map-injection : ∀ (f : A ↣ B) → List A ↣ List B
```

Comment:
As this is the propositional variant, I think we'd better name it `map-↣` and save `map-injection` for the setoid variety.